libs/shared/src/services/http-client/api.client.ts

```diff
@@ -14,47 +12,89 @@ export interface IPaginatedResponse<T = unknown> {
 }
 
 export class HttpClient {
-  private axiosClient: AxiosInstance;
+  private backendUrl: string;
+  private apiVersion = 'v1';
 
-  constructor(private backendUrl: string) {
-    this.axiosClient = axios.create({
-      baseURL: backendUrl + '/v1',
-    });
+  constructor(backendUrl: string) {
+    this.backendUrl = `${backendUrl}/${this.apiVersion}`;
   }
 
+  private headers: Record<string, string> = {
+    'Content-Type': 'application/json',
+  };
+
   setAuthorizationToken(token: string) {
-    this.axiosClient.defaults.headers.common.Authorization = `Bearer ${token}`;
+    this.headers.Authorization = `Bearer ${token}`;
   }
 
   disposeAuthorizationToken() {
-    delete this.axiosClient.defaults.headers.common.Authorization;
+    delete this.headers.Authorization;
   }
 
   async getFullResponse(url: string, params?: IParamObject) {
-    return await this.axiosClient
-      .get(url, {
-        params,
-      })
-      .then((response) => response.data);
+    const response = await fetch(this.backendUrl + url + this.getQueryString(params), {
+      headers: this.headers,
+    });
+
+    return await response.json();
   }
 
   async get(url: string, params?: IParamObject) {
-    return await this.axiosClient
-      .get(url, {
-        params,
-      })
-      .then((response) => response.data.data);
+    const response = await fetch(this.backendUrl + url + this.getQueryString(params), {
+      headers: this.headers,
+    });
+    const data = await response.json();
+
+    return data.data;
   }
 
   async post(url: string, body = {}) {
-    return await this.axiosClient.post(url, body).then((response) => response.data.data);
+    const response = await fetch(this.backendUrl + url, {
+      method: 'POST',
+      headers: this.headers,
+      body: JSON.stringify(body),
+    });
+    const data = await response.json();
+
+    return data.data;
   }
 
   async patch(url: string, body = {}) {
-    return await this.axiosClient.patch(url, body).then((response) => response.data.data);
+    const response = await fetch(this.backendUrl + url, {
+      method: 'PATCH',
+      headers: this.headers,
+      body: JSON.stringify(body),
+    });
+    const data = await response.json();
+
+    return data.data;
   }
 
   async delete(url: string, body = {}) {
-    return await this.axiosClient.delete(url, body).then((response) => response.data.data);
+    const response = await fetch(this.backendUrl + url, {
+      method: 'DELETE',
+      headers: this.headers,
+      body: JSON.stringify(body),
+    });
+    const data = await response.json();
+
+    return data.data;
+  }
+
+  private getQueryString(params?: IParamObject) {
```

Comment:
@BiswaViraj we can use [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) API that works in all JS runtimes.

Reply:
Tried using the URLSearchParams but it throws an typescript error due to the interface IParamObject.